src/FreeImageBmp.cpp

```diff
@@ -13,9 +14,13 @@ FreeImageBmp::FreeImageBmp(FreeImageBmp&& other) : fiBitmap(other.fiBitmap)
 	other.fiBitmap = nullptr;
 }
 
-FreeImageBmp::FreeImageBmp(int width, int height, int bpp) : 
+FreeImageBmp::FreeImageBmp(int width, int height, unsigned bpp) : 
 	fiBitmap(FreeImage_Allocate(width, height, bpp)) 
 {
+	if (bpp > INT_MAX) {
+		throw std::runtime_error("Bpp of " + std::to_string(bpp) + " is too large");
+	}
```

Comment:
Ahh right. Good eye pointing out that initialization. Could be a possible memory leak there when an exception is thrown, since the `fiBitmap` in not an RAII object. Might need to free the `fiBitmap` resource within the `if` body, before the `throw`.